src/Analyzer/Resolve/IdentifierLookup.h

```diff
@@ -166,19 +172,27 @@ struct IdentifierResolveResult
 
         return buffer.str();
     }
+
+private:
+    bool check_invariant() const noexcept
+    {
+        return (resolved_identifier == nullptr) == (resolve_place == IdentifierResolvePlace::NONE);
+    }
 };
 
 struct IdentifierResolveState
 {
-    IdentifierResolveResult resolve_result;
-    bool cyclic_identifier_resolve = false;
+    size_t count = 1;
 };
 
-struct IdentifierResolveSettings
+struct IdentifierResolveContext
 {
     /// Allow to check join tree during identifier resolution
     bool allow_to_check_join_tree = true;
 
+    /// Allow to check aliases during identifier resolution
+    bool allow_to_check_aliases = true;
```

Comment:
Can you please add some clarification in which case we would like to disable aliases (in the comment).